frontend/src/metabase/query_builder/selectors.js

```diff
@@ -1059,6 +1060,12 @@ export const getSubmittableQuestion = (state, question) => {
   const resultsMetadata = getResultsMetadata(state);
   const isResultDirty = getIsResultDirty(state);
 
+  if (question.type() === "model") {
+    resultsMetadata.columns = ModelIndexes.actions.cleanIndexFlags(
+      resultsMetadata.columns,
+    );
+  }
```

Comment:
This part should have been moved in #39147. It worked because it mutated data in redux store.